hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -242,18 +241,18 @@ class ColumnStats {
     records.forEach((record) -> {
       // For each column (field) we have to index update corresponding column stats
       // with the values from this record
-      targetFields.forEach(field -> {
-        ColumnStats colStats = allColumnStats.computeIfAbsent(field.name(), ignored -> new ColumnStats());
-        Schema fieldSchema = getNestedFieldSchemaFromWriteSchema(recordSchema, field.name());
+      targetFields.forEach(fieldNameFieldPair -> {
+        ColumnStats colStats = allColumnStats.computeIfAbsent(fieldNameFieldPair.getKey(), ignored -> new ColumnStats());
+        Schema fieldSchema = fieldNameFieldPair.getValue().schema();
```

Comment:
for readability - we can extract pair's key and value to separate fieldName and fieldSchema variables and use that below everywhere.